src/UI/SignTransactionsModals/index.tsx

```diff
@@ -71,39 +58,55 @@ export const SignTransactionsModals = ({
     verifyReceiverScam
   };
 
-  if (signError || hasTransactions) {
-    switch (loginMethod) {
-      case LoginMethodsEnum.ledger:
-        return CustomConfirmScreens?.Ledger ? (
-          <CustomConfirmScreens.Ledger {...signProps} />
-        ) : (
-          <SignWithLedgerModal {...signProps} />
-        );
-
-      case LoginMethodsEnum.walletconnect:
-      case LoginMethodsEnum.walletconnectv2:
-        return CustomConfirmScreens?.WalletConnect ? (
-          <CustomConfirmScreens.WalletConnect {...signProps} />
-        ) : (
-          <SignWithWalletConnectModal {...signProps} />
-        );
+  const ConfirmScreens: CustomConfirmScreensType = {
+    Ledger: CustomConfirmScreens?.Ledger ?? SignWithLedgerModal,
+    WalletConnect:
+      CustomConfirmScreens?.WalletConnect ?? SignWithWalletConnectModal,
+    Extension: CustomConfirmScreens?.Extension ?? SignWithExtensionModal,
+    Extra: CustomConfirmScreens?.Extra
+  };
 
-      case LoginMethodsEnum.extension:
-        return CustomConfirmScreens?.Extension ? (
-          <CustomConfirmScreens.Extension {...signProps} />
-        ) : (
-          <SignWithExtensionModal {...signProps} />
-        );
+  const shouldShowTransactionStatusToast =
+    Boolean(signError) || Boolean(cancelTransactionsMessage);
 
-      case LoginMethodsEnum.extra:
-        return CustomConfirmScreens?.Extra ? (
-          <CustomConfirmScreens.Extra {...signProps} />
-        ) : null;
+  const renderScreen = useCallback(
+    (Screen?: ScreenType) => {
```

Comment:
This logic seems a bit crowded, a lot of features into a couple of components with a lot of props. Is there a way we can simplify this rendering, so it's easier to read?

Reply:
simplified